components/server-side-render/index.js

```diff
@@ -12,6 +12,7 @@ import {
 } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 import apiRequest from '@wordpress/api-request';
+import httpBuildQuery from 'http-build-query';
```

Comment:
Should we use `querystring` instead? It's already bundled in webpack and used by several packages in Gutenberg?

Reply:
@youknowriad PHP has its own unique format for query strings of arrays/objects.

Here's a rough example showing how each system encodes the query string differently:

```
qs = require('querystring');
hbq = require('http-build-query');

const obj = { "array": [ "a", "b", "c" ] };

qs.stringify(obj);
// array=a&array=b&array=c

hbq(obj);
// array[0]=a&array[1]=b&array[2]=c
```

Because `http-build-query` is just re-implementing PHP's native `http_build_query` function, it's transforming into the format that PHP can properly decode. If the `querystring` version of that object was passed onto PHP, it would be decoded as `array( 'array' => 'c' )` and drop the other values in the array.

Reply:
I tested `querystring-es3` locally but it doesn't pass our tests:

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/36432/41598316-b6e69676-7384-11e8-9c36-583a5a09a698.png">

It appears the project is relatively abandoned too: https://github.com/SpainTrain/querystring-es3

Reply:
I'm pretty sure there's an option in `querystring` to do the exact same thing.

Reply:
@youknowriad I don't believe it does. I also looked at `querystringify` which is also included in Gutenberg and also doesn't support this format.

Reply:
Ok so discussed in Slack, it's fine to keep `http-build-query` but I'd appreciate if we don't rush into merging when there's unresolved comments.

Reply:
> I'd appreciate if we don't rush into merging when there's unresolved comments.

At the time I merged, it wasn't clear there were unresolved comments.